python/pyarrow/tests/test_flight.py

```diff
@@ -483,6 +495,51 @@ def sending_headers(self):
         }
 
 
+class MultiHeaderClientMiddlewareFactory(ClientMiddlewareFactory):
+    """Test sending/receiving multiple (binary-valued) headers."""
+
+    def __init__(self):
+        self.last_headers = {}
```

Comment:
This isn't ever used, right?

Reply:
It is used, but it is unclear, so I've added an explanation of where and how it's used.